Reflection/Internal/RSGroundObject.simba

```diff
@@ -70,6 +80,141 @@ begin
   RFreeObject(R_EIOS, Items);
 end;
 
+(*
+RSGroundObject.GetTileByID
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Returns all tiles that contain a specific object ID.
+
+Example:
+
+  Writeln(RSGroundObject.GetTileByID(526));
+*)
+Function RSGroundObject.GetTileByID(itemID: Int32): Array of RSTile; static;
+var
+  Tile: RSTile;
+  Plane: Int32;
+  BaseX, BaseY: Int32;
+  MinX, MinY, MaxX, MaxY: Int32;
+  X, Y: Int32;
+  Player: RSPlayer;
+
+  Items: Pointer;
+  ItemsSize: SizeUInt;
+  NodeDeque: RSNodeDeque;
+  Head, Curr: RSNode;
+  Garbage: RSTYpeArray;
+
+  ID: Int32;
+begin
+  Player := RSPlayer.Me;
+  Tile := Player.Tile;
+  Player.Free;
+
+  BaseX := RSClient.BaseX;
+  BaseY := RSClient.BaseY;
+  Plane := RSClient.Plane;
+
+  MinX := BaseX;
+  MinY := BaseY;
+  MaxX := BaseX + 104;
+  MaxY := BaseY + 104;
+
+  ItemsSize := 0;
+  Items := RGetArray(R_EIOS, nil, ItemsSize, Client_GroundItems);
+  if ItemsSize = 0 then
+    Exit;
+
+  for Y := MinY to MaxY - 1 do
+  begin
+    for X := MinX to MaxX - 1 do
+    begin
+      NodeDeque.ref := RGetObjectArray3D(R_EIOS, Items, Plane, X - BaseX, Y - BaseY);
+      if NodeDeque.ref <> nil then
+      begin
+        Head := NodeDeque.Head;
+        Curr := Head.Next;
+        Garbage := [NodeDeque, Head, Curr];
+
+        while not RObjectsEqual(R_EIOS, Head.ref, Curr.ref) do
+        begin
+          ID := RGetInt(R_EIOS, Curr.ref, ITEM_ID);
+          if (itemID <> ID) then
+            break();
```

Comment:
This won't work? It shouldn't break out of the loop. It should instead loop all the way and just add the ones it found to the result.